gmso/tests/test_potential_templates.py

```diff
@@ -0,0 +1,129 @@
+                                                                                ' c4 * cos(phi)**4 + c5 * cos(phi)**5')
+        assert ryckaert_bellemans_torsion_potential.independent_variables == {sympy.sympify('phi')}
+
+    def test_harmonic_torsion_potential(self, templates):
+        harmonic_torsion_potential = templates['HarmonicTorsionPotential']
+        assert harmonic_torsion_potential.name == 'HarmonicTorsionPotential'
+        assert harmonic_torsion_potential.expression == sympy.sympify('0.5 * k * (phi - phi_eq)**2')
+        assert harmonic_torsion_potential.independent_variables == {sympy.sympify('phi')}
+
+    def test_harmonic_improper_potential(self, templates):
+        harmonic_improper_potential = templates['HarmonicImproperPotential']
+        assert harmonic_improper_potential.name == 'HarmonicImproperPotential'
+        assert harmonic_improper_potential.expression == sympy.sympify('0.5 * k * (phi - phi_eq)**2')
+        assert harmonic_improper_potential.independent_variables == {sympy.sympify('phi')}
+
+    def test_harmonic_bond_potential(self, templates):
+        harmonic_bond_potential = templates['HarmonicBondPotential']
+        assert harmonic_bond_potential.name == 'HarmonicBondPotential'
+        assert harmonic_bond_potential.expression == sympy.sympify('0.5 * k * (r-r_eq)**2')
+        assert harmonic_bond_potential.independent_variables == {sympy.sympify('r')}
+
+    def test_harmonic_angle_potential(self, templates):
+        harmonic_bond_potential = templates['HarmonicAnglePotential']
+        assert harmonic_bond_potential.name == 'HarmonicAnglePotential'
+        assert harmonic_bond_potential.expression == sympy.sympify('0.5 * k * (theta-theta_eq)**2')
+        assert harmonic_bond_potential.independent_variables == {sympy.sympify('theta')}
+
+    def test_buckingham_potential(self, templates):
+        buckingham_potential = templates['BuckinghamPotential']
+        assert buckingham_potential.name == 'BuckinghamPotential'
+        assert buckingham_potential.expression == sympy.sympify('a*exp(-b*r) - c*r**-6')
+        assert buckingham_potential.independent_variables == sympy.sympify({'r'})
+
+    def test_save_templates(self, templates):
+        PotentialTemplates.save_potential_template('TestTemplate',
+                                                   {
+                                                       'name': 'TestTemplate',
+                                                       'expression': 'a+b+c',
+                                                       'independent_variables': 'c,a'
+                                                   },
+                                                   update=True,
+                                                   user_template=False
+                                                   )
+
+        assert os.path.exists(os.path.join(JSON_DIR, 'TestTemplate.json'))
```

Comment:
Is there a way to ensure that when running these tests, the test json files are cleaned up?

Reply:
Would this be achieved with a tempdir fixture?

Reply:
If there isn't a fixture this is a code snippet I've been using to contain files I'm creating in tests:

```python
import os
import tempfile
import contextlib
import shutil

@contextlib.contextmanager
def temporary_cd():
    tmp_dir = tempfile.mkdtemp()
    prev_dir = os.getcwd()
    os.chdir(os.path.abspath(tmp_dir))
    try:
        yield
    finally:
        os.chdir(prev_dir)
        shutil.rmtree(tmp_dir)
```

Reply:
I think you can do a bit better by using the temporary directory as a context manager. I also think you can skip the `abspath` this way:
```python
import os
import tempfile
import contextlib

@contextlib.contextmanager
def temporary_cd():
    prev_dir = os.getcwd()
    with tempfile.TemporaryDirectory() as tmp_dir:
        try:
            os.chdir(tmp_dir)
            yield
        finally:
            os.chdir(prev_dir)
```